packages/flutter_tools/lib/src/context_runner.dart

```diff
@@ -171,6 +172,11 @@ Future<T> runInContext<T>(
         flutterProjectFactory: globals.projectFactory,
         client: globals.httpClientFactory?.call() ?? HttpClient(),
       ),
+      CustomDevicesConfig: () => CustomDevicesConfig(
+        fileSystem: globals.fs,
+        logger: globals.logger,
+        platform: globals.platform,
+      )..ensureFileExists(),
```

Comment:
rather that do this here (because it might be easy to miss), I would run this function in the constructor

Reply:
okay, fixed it